app/utils/timeline-utils.ts

```diff
@@ -144,6 +145,54 @@ function getDirectionHistory(req: Request): any[] {
   }
 }
 
+function getUpdateTribunalDecisionHistory(req: Request, ftpaSetAsideFeatureEnabled: boolean): any[] {
+  if (isUpdateTribunalDecideWithRule31(req, ftpaSetAsideFeatureEnabled)) {
+
+    let originalTribunalDecision = req.session.appeal.isDecisionAllowed && req.session.appeal.isDecisionAllowed.toLowerCase() || null;
+    let updatedAppealDecision = req.session.appeal.updatedAppealDecision && req.session.appeal.updatedAppealDecision.toLowerCase() || null;
+    let timelineText = '';
+
+    if (originalTribunalDecision === 'allowed' && updatedAppealDecision === 'dismissed') {
+      timelineText = i18n.pages.overviewPage.timeline.updateTribunalDecision.fromAllowedToDismissedText;
+    } else if (originalTribunalDecision === 'dismissed' && updatedAppealDecision === 'allowed') {
+      timelineText = i18n.pages.overviewPage.timeline.updateTribunalDecision.fromDismissedToAllowedText;
+    } else {
+      return [];
+    }
+
+    let latestUpdateTribunalDecisionHistory = req.session.appeal.history
+      .filter(history => history.id === Events.UPDATE_TRIBUNAL_DECISION.id)
+      .sort((a: any, b: any) => b.dateObject - a.dateObject)[0];
+
+    return [{
+      date: moment(latestUpdateTribunalDecisionHistory.createdDate).format('DD MMMM YYYY'),
+      dateObject: new Date(latestUpdateTribunalDecisionHistory.createdDate),
+      text: timelineText || null
+    }];
+  } else {
+    return [];
+  }
+}
+
+function getUpdateTribunalDecisionDocumentHistory(req: Request, ftpaSetAsideFeatureEnabled: boolean): any[] {
+  if (isUpdateTribunalDecideWithRule31(req, ftpaSetAsideFeatureEnabled) && req.session.appeal.decisionAndReasonDocsUpload) {
+
+    const coverLetterDocument = req.session.appeal.finalDecisionAndReasonsDocuments.find(doc => doc.tag === 'updatedDecisionAndReasonsCoverLetter');
+
+    return [{

```

Comment:
As discussed, lets review this together. See if we can use the UPDATE_TRIBUNAL_DECISION event.